clients/src/main/java/org/apache/kafka/common/Node.java

```diff
@@ -28,20 +28,60 @@
     private final String host;
     private final int port;
     private final String rack;
+    private final boolean privileged;
 
     // Cache hashCode as it is called in performance sensitive parts of the code (e.g. RecordAccumulator.ready)
     private Integer hash;
 
+    /**
+     * Creates a new Node representing a Kafka broker
+     * @param id Identifier of the Node
+     * @param host Hostname used to connect
+     * @param port Port used to connect
+     */
     public Node(int id, String host, int port) {
-        this(id, host, port, null);
+        this(id, host, port, null, false);
     }
 
+    /**
+     * Creates a new Node representing a Kafka broker
+     * @param id Identifier of the Node
+     * @param host Hostname used to connect
+     * @param port Port used to connect
+     * @param rack Rack name
+     */
     public Node(int id, String host, int port, String rack) {
+        this(id, host, port, rack, false);
+    }
+
+    /**
+     * Creates a new Node representing a Kafka broker
+     * @param id Identifier of the Node
+     * @param host Hostname used to connect
+     * @param port Port used to connect
+     * @param privileged If true, messages from this node will be allocated directly on the heap when the MemoryPool 
+     * is full
+     */
+    public Node(int id, String host, int port, boolean privileged) {
```

Comment:
This API change is covered by the KIP

Reply:
One new constructor is in the KIP, but do we still think it is a good idea? The notion of a privileged connection is only relevant to the internal implementation, so it would be unfortunate for it to leak into the public API.

On a related note, we were discussing whether it makes sense to allocate JoinGroup/SyncGroup using the buffer pool. I'm wondering if that suggests we should be able to control this allocation at the request level instead of the connection level. If we were doing it at the request level, then we could add an additional flag in `ClientRequest`. Then we wouldn't need the changes to `Node` here.

Reply:
@hachikuji 

@edoardocomar and I have thought about your suggestion and we think we could get this to work without touching `Node`. The idea would be to rely on the request/response guaranteed ordering and match the incoming response with a pending request.

We could inject into `KafkaChannel`s an interface that would help it determine whether or not to read a response in the memoryPool - based on the apikey of the associated request.
The Consumer implementation of the interface would simply `peekLast()` the `inFlightRequests` collection held by `NetworkClient` and return whether the matching request requires allocation into the pool (i.e. whether it is a FETCH)

That way, allocation would be determined by the apikey of the request instead of the node.

What do you think?